pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -116,6 +116,11 @@ type ApiServerResource struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector `json:"labelSelector"`
 
+	// ControllerSelector restricts this source to objects with the selected controller kind.
```

Comment:
I would prefer if this includes something about owner references. Unless this is common phrasing in K8s. (I got confused at first by thinking about a controller as the code doing reconcile loops. Something like:

ControllerSelector restricts this source to objects with a controlling owner reference of the specified kind.

Reply:
yes much better, thanks :-)